app/cronjob/trailingTrade/step/place-buy-order.js

```diff
@@ -87,43 +101,87 @@ const execute = async (logger, rawData) => {
   }
 
   if (freeBalance < parseFloat(minNotional)) {
-    data.buy.processMessage = `Do not place a buy order as not enough ${quoteAsset} to buy ${baseAsset}.`;
+    data.buy.processMessage =
+      `Do not place a buy order for the grid trade #${humanisedGridTradeIndex} ` +
+      `as not enough ${quoteAsset} to buy ${baseAsset}.`;
     data.buy.updatedAt = moment().utc();
 
     return data;
   }
 
-  const stopPrice = roundDown(currentPrice * stopPercentage, pricePrecision);
-  const limitPrice = roundDown(currentPrice * limitPercentage, pricePrecision);
+  const stopPrice = roundDown(
```

Comment:
In the initial development, there was some case, lodash floor was not working as expected. So I had to use a different approach to get. And there were not many test cases to verify the result. 

Since I have various test cases to verify the result, I changed back to `_.floor` and it seems working fine. 
I will change to `_.floor`.

Thanks for the suggestion!